### PR TITLE
[#2394] feat(IT): Add metalake page to web e2e test

### PR DESCRIPTION
**Pull request by ch3yne**
### What changes were proposed in this pull request?

Add web e2e test to homepage.

### Why are the changes needed?

Fix: #2394 

### Does this PR introduce _any_ user-facing change?

N/A

### How was this patch tested?

N/A

### CODE REVIEW

**Comment by xunliu** [comment]
@ch3yne Please fix Github Action failed.
You can execute `./gradew build` in gravitino root directory.

**Review by xunliu — COMMENTED** [review]
@LauraXia123 Please help review this PR, Thanks!

File: integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/MetalakePageTest.java [hunk 1]
```diff
@@ -4,14 +4,133 @@
...
+
+  @Test
+  @Order(1)
+  public void testCreateMetalake() {
+    metalakePage.createMetalakeAction("test", "test", true);
+
+    boolean status = metalakePage.verifyIsCreatedMetalake();
+
+    if (status) {
+      LOG.info("create metalake successful");
```
Comment (jerryshao) [inline integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/MetalakePageTest.java:33]: We don't have to do the log in  java test. Just check the return value is expected or not is enough. @xunliu please help @ch3yne to make this test more java-ish.

File: integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/Pages/MetalakePage.java [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+  public boolean verifyCreateMetalake(String name) {
+    try {
+      String rowPath = "//div[@data-id='" + name + "']";
+      WebElement createdMetalakeRow = driver.findElement(By.xpath(rowPath));
+      boolean isRow = createdMetalakeRow.isDisplayed();
+
+      String linkPath = "//div[@data-field='name']//a[@href='/ui/metalakes?metalake=" + name + "']";
+      WebElement createdMetalakeLink = driver.findElement(By.xpath(linkPath));
+      boolean isLink = createdMetalakeLink.isDisplayed();
+      boolean isText = Objects.equals(createdMetalakeLink.getText(), name);
```
Comment (xunliu) [inline integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/Pages/MetalakePage.java:147]: Please add some comments to describe these codes.

File: integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/Pages/MetalakePage.java [hunk 3]
```diff
@@ -123,16 +123,27 @@ public void setMetalakeProps(int index, String key, String value) {
...
-            .ignoring(ElementNotInteractableException.class);
-    wait.until(
-        d -> {
-          findElementByLink(name).isDisplayed();
-          return true;
-        });
+    int retry = 0;
+    int sleepTimeMillis = 1_000;
+
+    while (retry++ < 3) {
```
Comment (xunliu) [inline integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/Pages/MetalakePage.java:129]: The 3 second too short, Please change to `5`.

File: integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/Pages/MetalakePage.java [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+      return Objects.equals(editedMetalakeLink.getText(), name);
+    } catch (Exception e) {
+      return false;
+    }
+  }
+
+  public boolean verifyShowMetalakeDetails() {
+    try {
+      metalakeDetailsDrawer.isDisplayed();
+      String drawerVisible = metalakeDetailsDrawer.getCssValue("visibility");
```
Comment (xunliu) [inline integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/Pages/MetalakePage.java:185]: It's too sketchy to just check if it's visible or not.
Don't we need check value on each element?

**Comment by LauraXia123** [comment]
no comments for me

File: integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/MetalakePageTest.java [hunk 5]
```diff
@@ -74,7 +84,7 @@ public void testCreateMultipleMetalakes() {
 
     for (int i = 0; i < towPagesCount; i++) {
```
Comment (xunliu) [inline integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/MetalakePageTest.java:85]: What mean is `towPagesCount`?

File: integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/Pages/MetalakePage.java [hunk 6]
```diff
@@ -265,6 +279,11 @@ public boolean verifyLinkToCatalogsPage(String name) {
       return nameLink.isDisplayed() && isUrl;
     } catch (Exception e) {
       return false;
+    } finally {
+      // Back to homepage
+      String xpath = "//button[@data-refer='back-home-btn']";
+      WebElement backHomeBtn = driver.findElement(By.xpath(xpath));
```
Comment (xunliu) [inline integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/Pages/MetalakePage.java:286]: Whether need define `backHomeBtn` to be a MetalakePage Class vaiable?

File: integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/utils/AbstractWebIT.java [hunk 7]
```diff
@@ -15,7 +15,7 @@
 public class AbstractWebIT extends AbstractIT {
   protected static final Logger LOG = LoggerFactory.getLogger(AbstractWebIT.class);
   protected static WebDriver driver;
-  protected static final long MAX_IMPLICIT_WAIT = 60;
+  protected static final long MAX_IMPLICIT_WAIT = 10;
```
Comment (xunliu) [inline integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/utils/AbstractWebIT.java:18]: Why need reduce wait time？

Comment (ch3yne) [reply integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/utils/AbstractWebIT.java:18]: When searching for elements, this implicit wait time is too long, which will increase the time it takes for testing.

File: integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/Pages/MetalakePage.java (hunk 6 above)
Comment (ch3yne) [reply integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/Pages/MetalakePage.java:286]: I added the backHomeBtn Class in CatalogsPage.java, I will modify when the PR is submited.

File: integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/MetalakePageTest.java (hunk 5 above)
Comment (ch3yne) [reply integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/MetalakePageTest.java:85]: The default counts of one page is 10, I need create over 11 counts of metalakes to make the `Go to next page` available.

File: integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/Pages/MetalakePage.java [hunk 8]
```diff
@@ -0,0 +1,295 @@
...
+    valueInput.sendKeys(value);
+  }
+
+  public boolean checkIsErrorName() {
+    String xpath = "//div[@data-refer='metalake-name-field']";
+    WebElement nameField = findElementByXPath(xpath);
+    List<WebElement> errorText =
+        nameField.findElements(By.xpath("//div[contains(@class, 'Mui-error')]"));
+
+    return !errorText.isEmpty();
```
Comment (xunliu) [inline integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/Pages/MetalakePage.java:145]: Please use `Strings.isEmpty()` function.

Comment (ch3yne) [reply integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/Pages/MetalakePage.java:146]: This is the element function.

**Review by xunliu — APPROVED** [review]
LGTM

**Comment by xunliu** [comment]
@ch3yne Please remember fix #2512 when you have time.

**Pull request merged**